### PR TITLE
Add eager support for aten::equal and expand the test coverage around…

### PR DESCRIPTION
**Pull request by WilBrady**
**Description**: Do not review. This is a variety of changes that I will be breaking up into smaller PRs

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: orttraining/orttraining/eager/ort_aten.cpp [hunk 1]
```diff
@@ -682,6 +682,69 @@ at::Tensor& out) {
...
+      &at::native::cpu_fallback,
+      ATEN_OP(equal)>::call(self, other);
+  }
+
+  auto& invoker = GetORTInvoker(self.device());
+
+  auto ort_input_self = create_ort_value(invoker, self);
+  auto ort_input_other = create_ort_value(invoker, other);
+
+  auto* ort_tensor_self = ort_input_self.GetMutable<onnxruntime::Tensor>();
```
Comment (jamill) [inline orttraining/orttraining/eager/ort_aten.cpp:706]: Can these be switched to the non-mutable version?

File: orttraining/orttraining/eager/test/ort_ops.py [hunk 2]
```diff
@@ -120,9 +120,24 @@ def test_min(self):
...
+        ort_a = cpu_a.to(device)
+        cpu_b = torch.Tensor([1.0, 1.5])
+        ort_b = cpu_b.to(device)
+        cpu_c = torch.Tensor([1.0, 1.8])
+        ort_c = cpu_c.to(device)
+        cpu_d = torch.Tensor([1.0, 1.5, 2.1])
+        ort_d = cpu_d.to(device)
+        cpu_e = torch.Tensor([[1.0, 1.5]])
+        ort_e = cpu_e.to(device)
+        assert torch.equal(cpu_a, cpu_b)
```
Comment (jamill) [inline orttraining/orttraining/eager/test/ort_ops.py:133]: suggestion: did you want to include comment about what aspects / properties each of the different tests are excercising?

**Review by jamill — APPROVED** [review]
Changes look good - please do update the PR description before marking as ready.

File: orttraining/orttraining/eager/ort_aten.cpp (hunk 1 above)
Comment (WilBrady) [reply orttraining/orttraining/eager/ort_aten.cpp:713]: posting a new pr for just equal and i'll make this change there. Thank you @jamill

**Comment by WilBrady** [comment]
All work in the PR has been done via smaller PRs. Closing this as it is obsolete now.

**Pull request closed without merging**